src/runtime/virtcontainers/api_test.go

```diff
@@ -170,6 +170,7 @@ func TestCreateSandboxKataAgentSuccessful(t *testing.T) {
 
 	url, err := mock.GenerateKataMockHybridVSock()
 	assert.NoError(err)
+	defer os.RemoveAll(filepath.Dir(strings.TrimPrefix(url, "mock://")))
```

Comment:
Much piecemeal processing spans multiple modules, it is recommended to encapsulate Mock hybrid vsock.

For example, just like this, the mock side:

```golang
type MockHybridVSock struct {
	dir    string
	socket string
}

func (hvs MockHybridVSock) clean() {
	os.RemoveAll(hvs.dir)
}

func (hvs MockHybridVSock) address() string {
	return hvs.socket
}

func GenerateKataMockHybridVSock() (MockHybridVSock, error) {
	dir, err := ioutil.TempDir("", "kata-mock-hybrid-vsock-test")
	if err != nil {
		return MockHybridVSock{}, err
	}

	m := MockHybridVSock{
		dir:    dir,
		socket: fmt.Sprintf(testKataMockHybridVSockURLTempl, dir),
	}

	return m, nil
}
```

And the user side:

```golang
	hvSocket, err := mock.GenerateKataMockHybridVSock()
	assert.NoError(err)
	url := hvSocket.address()
	defer hvSocket.clean()
```

Reply:
Thanks for the  advice, and how about adding a function to clean it up,  because  I don't want to change the definition of the  GenerateKataMockHybridVSock() function.  Add a clean function such as:
```
func RemoveKataMockHybridVSock(sockAddress string) error {
    var sockPrefix = "mock://"

    if !strings.HasPrefix(sockAddress, sockPrefix) {
        return fmt.Errorf("Invalid socket address: %s", sockAddress)
    }

    sockPath := strings.TrimPrefix(sockAddress, sockPrefix)

    return os.RemoveAll(path.Dir(sockPath))
}
```
and use  mock. RemoveKataMockHybridVSock(url)  in  the test code.